src/admin/shared/components/ContentPicker/ContentPicker.tsx

```diff
@@ -17,88 +17,103 @@ import {
+	// filter available options for the type picker
+	const typeOptions = filterTypes(CONTENT_TYPES, allowedTypes);
+
+	// apply initial type from `initialValues`, default to first available type
+	const currentTypeObject = typeOptions.find(type => type.value === get(initialValues, 'type'));
+	const [selectedType, setSelectedType] = useState<PickerTypeOption<ContentPickerType>>(
 		currentTypeObject || typeOptions[0]
 	);
-	const [options, setOptions] = useState<PickerSelectItem[]>([]);
-	const [input, setInput] = useState<string>(
-		setInitialInput(currentTypeObject, currentSelection)
-	);
-	const [currentItem, setCurrentItem] = useState<ValueType<PickerItem> | null>(
-		setInitialItem(options, currentSelection)
-	);
 
-	// inflate content picker
+	// available options for the item picker.
+	const [itemOptions, setItemOptions] = useState<PickerSelectItem[]>([]);
+
+	// selected option, keep track of wether initial item from `initialValues` has been applied
+	const [selectedItem, setSelectedItem] = useState<ValueType<PickerItem>>();
+	const [hasAppliedInitialItem, setHasAppliedInitialItem] = useState<boolean>(false);
+
+	// apply initial input if INPUT-based type, default to ''
+	const [input, setInput] = useState<string>(setInitialInput(currentTypeObject, initialValues));
+
+	// inflate item picker // TODO: type
 	const inflatePicker = (keyword: string | null, callback: any) => {
-		if (currentType && !!currentType.fetch) {
-			currentType
+		if (selectedType && !!selectedType.fetch) {
+			selectedType
 				.fetch(keyword, 20)
-				.then((items: any) => {
-					callback((items as any) || []);
-
-					setCurrentItem(null);
-				})
-				.catch((error: any) => {
-					console.error('[Content Picker] - Failed to inflate.', error);
-					ToastService.danger(
-						t(
-							'admin/content/components/content-picker/content-picker___het-ophalen-van-de-content-items-is-mislukt'
-						),
-						false
-					);
-				});
+				.then((items: any) => callback((items as any) || [])) // TODO: type
+				.catch(handleInflationError);
 		}
 	};
 
+	// handle error during inflation of item picker // TODO: type
+	const handleInflationError = (error: any) => {
+		console.error('[Content Picker] - Failed to inflate.', error);
+		ToastService.danger(
+			t(
+				'admin/content/components/content-picker/content-picker___het-ophalen-van-de-content-items-is-mislukt'
+			),
+			false
+		);
+	};
+
+	// when selecting a type, reset `selectedItem` and retrieve new item options
 	useEffect(() => {
-		inflatePicker(null, setOptions);
-	}, [currentType]); // eslint-disable-line
+		inflatePicker(null, setItemOptions);
+	}, [selectedType]);
+
+	// during the first update of `itemOptions`, set the initial value of the item picker
+	useEffect(() => {
+		if (itemOptions.length && !hasAppliedInitialItem) {
+			setSelectedItem(setInitialItem(itemOptions, initialValues));
+			setHasAppliedInitialItem(true);
+		}
+	}, [itemOptions]);
```

Comment:
`React Hook useEffect has missing dependencies: 'hasAppliedInitialItem' and 'initialValues'`